### PR TITLE
Bogavril/x client os

### LINKED ISSUE #2517: [Bug] MSAL not sending correct OS version string on .NET classic
msal 4.29

MSAL sends x-client-os header with version string. THis is used by the server for device auth.
On .NET classic, this functionality is broken, as we mostly just send "6.2.5900" due to a bug in .NET fwk (see https://stackoverflow.com/questions/33328739/system-environment-osversion-returns-wrong-version)

This fixes the issue by send the correct OS description, e.g. "Windows 10", "Windows Server 2016" etc.

### PR DESCRIPTION
**Pull request by bgavrilMS**
Send correct x-client-os OS info on .NET in header

Fixes #2517

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/uap/UapPlatformProxy.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ protected override string InternalGetOperatingSystem()
         {
             // In WinRT, there is no way to reliably get OS version. All can be done reliably is to check
             // for existence of specific features which does not help in this case, so we do not emit OS in WinRT.
-            return null;
+            return "Windows 10";
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/uap/UapPlatformProxy.cs:110]: What is the reason for this change? Maybe should update the comment to match.

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Platforms/uap/UapPlatformProxy.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/uap/UapPlatformProxy.cs:108]: Removed the comment. We should always send the OS as ESTS makes some decisions based on it,.
